### PR TITLE
Locale-invariant parsing and formatting

### PR DESCRIPTION
**Pull request by dkhalanskyjb**
The same thing as #251 (except for small tidying up here and there), with the commit history rewritten to be more manageable. @ilya-g, does this commit history look ok?

### CODE REVIEW

File: core/common/src/format/DateTimeFormat.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+     */
+    ZERO,
+
+    /**
+     * Pad with spaces.
+     */
+    SPACE
+}
+
+internal fun Padding.toKotlinCode(): String = when (this) {
```
Comment (ilya-g) [inline core/common/src/format/DateTimeFormat.kt:74]: May be worth to move to `toKotlinCode.kt`

**Review by ilya-g — APPROVED** [review]
Ok, all is clear

File: core/common/src/format/DateTimeFormat.kt (hunk 1 above)
Comment (dkhalanskyjb) [reply core/common/src/format/DateTimeFormat.kt:74]: I tried to stick to the principle that the code expected to change should be kept together. For example, `DayOfWeekNames.toKotlinCode` is directly below `DayOfWeekNames`, the field specifications are right next to the directives involving the field, etc. WDYT?

Comment (ilya-g) [reply core/common/src/format/DateTimeFormat.kt:74]: ok

**Comment by Amejonah1200** [comment]
G.O.A.T. 🐐

**Pull request merged**